proposed/dotnet-bootstrapper/dotnet-bootstrapper-path.md

```diff
@@ -0,0 +1,130 @@
+
+> 📓 Technical Details: Update
+> Upon the action of step 5, we would need to take on the responsibility of updating these SDKs. Managing a daemon is resource-intensive, so it may not be optimal. SDL requirements specify a required update policy; we cannot hook into MU and IT administrators rely on administrative install behaviors. One approach is to mimic the update behavior of Visual Studio upon every interaction with `bootstrapper` by suggesting updates and asking the user to specify an update cadence when the new local mode is applied. IT administrators should be able to enforce a policy that requires a specific update cadence, and the provided update cadence should be compliant with SDL requirements, such as checking for updates on a daily cadence when the tool is used. Visual Studio would need to trigger this check as well to maintain compliance.
+
+#### Advantages and Disadvantages
+
+The advantage of this is that dotnet installations created by `bootstrapper` would work even if Visual Studio updates. Another advantage of changing the behavior based on a user action is that this change would not break existing installations of VS or existing code/tooling scenarios without user interaction. It's then possible to educate the user about their decision to change the PATH behavior.
+
+The disadvantage is that Visual Studio / other tooling would no longer be able to find its SDKs once this mode was enabled, so the SDK update that is included with VS would not take effect without additional action. The SDK included with VS is an administrative install, which is managed by MU. This means MU would also lose authority to update VS's SDKs; MU (Microsoft Update) and other related IT update mechanisms are tied to MSI or administrative installs, so there is no easy way to hook a local installation mechanism into that toolchain. This is addressed shortly.
+
+A disadvantage of this setup is that older .NET Installers that do not respect the setting would still interfere with `bootstrapper` by modifying the system `PATH`. (For example, the `pkg` installer uses `tee` which overwrites the `PATH` file rather than just appending to it.) While we can potentially service these installers, older versions would still cause this issue. Addressing this comprehensively would require modifying the application host or `dotnet.exe`, which is not achievable within the .NET 10 timeframe and would need to be targeted for .NET 11. The advantage is that the application host always runs on the latest version, which would enable it to handle older installers appropriately.
```

Comment:
This disadvantage doesn't apply as much as it may seem.  The installer component that sets the path is the one that lays out dotnet.exe and hostfxr, which updates in-place to the latest version.  So as long as any admin version of .NET 10 was installed, installing or updating an earlier version of .NET would not update this component and hence wouldn't modify the system PATH.

Reply:
I'm unsure of the technical aspects of this so I've reflected that in the doc, this is interesting

Reply:
Thanks for that context, I will update the document.

Reply:
> There is a separate installer for host FXR and the host. The host MSI installs dotnet.exe and the component responsible for the PATH.

Are these part of Runtime installer or SDK installer? For example, if I install ".NET Runtime" or ".NET Desktop Runtime" only, do I still get `dotnet` on PATH?

Can these PATH modifications affect non-SDK scenarios? If the PATH is modified as part Runtime installer, it is likely that there are applications out there that depend on dotnet.exe to be on the path, and switching the machine to local SDK mode may break them in various ways.

Reply:
>Can these PATH modifications affect non-SDK scenarios 

Yes, there are a lot of security boundaries we would have to consider w.r.t runtimes, such as another account relying on the system wide PATH for a runtime like described above.